src/foam/mlang/mlang.js

```diff
@@ -4702,6 +4711,54 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  package: 'foam.mlang',
+  name: 'If',
+  extends: 'foam.mlang.AbstractExpr',
+
+  properties: [
+    {
+      class: 'foam.mlang.predicate.PredicateProperty',
+      name: 'predicate',
+      javaFactory: 'return foam.mlang.MLang.TRUE;'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'valueIfTrue'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'valueIfFalse'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        if ( getPredicate().f(obj) ) {
+          return  getValueIfTrue() != null ?  getValueIfTrue().f(obj) : null;
+        } else {
+          return getValueIfFalse() != null ? getValueIfFalse().f(obj) : null;
+        }
+      `
+    },
+    {
+      name: 'toString',
+      type: 'String',
+      javaCode: `
+        var sb = new StringBuilder();
+        sb.append("If(")
+          .append("predicate:").append(String.valueOf(getPredicate()))
```

Comment:
String.valueOf() is already handled by overloaded append() methods. Also, since you can do this all in one statement, it's the exact same as you had just done return "If(predicate:" + getPredicate + ", valueIfTrue:" + ...